src/app/forget-password/reset/[token]/form/index.tsx

```diff
@@ -0,0 +1,186 @@
+import ResetPasswordRequest from '@/services/resetPasswordRequest';
+
+type PasswordValues = {
+  newPassword: string;
+  confirmNewPassword: string;
+};
+
+type FromProps = {
+  token: string;
+};
+
+const ResetPasswordForm = ({ token }: FromProps) => {
+  const router = useRouter();
+  const [isOpen, setIsOpen] = useState(false);
+  const [response, setResponse] = useState<CustomResponse>({
+    status: '',
+    title: '',
+    message: '',
+  });
+
+  const handleCloseModal = () => {
+    setIsOpen(false);
+  };
+
+  const onSubmit = async (values: PasswordValues) => {
+    try {
+      const { response } = await ResetPasswordRequest(token, values.newPassword);
+      if (response.status === 200) {
+        setResponse({
+          status: 'success',
+          title: 'Successful',
+          message: 'Password has been reset',
+        });
+
+        setIsOpen(true);
+
+        setTimeout(() => {
+          handleCloseModal();
+          router.push(Pages.SIGNIN);
+        }, 4000);
+      }
+    } catch (error: any) {
+      let message = 'An unknown error occurred.';
+
+      switch (error.response.status) {
+        case 400:
+          message = ErrorMsg.EXPIRED_TOKEN;
+          break;
+        case 500:
+          message = ErrorMsg.SERVER_ERROR;
+        default:
+          message = 'An unknown error occurred.';
+      }
+
+      setIsOpen(true);
+
+      setResponse({
+        status: 'error',
+        title: 'Reset Password Failed',
+        message,
+      });
+
+      setTimeout(() => {
+        handleCloseModal();
+      }, 3000);
+    }
+  };
+
+  const handleRedirect = () => {
+    router.push(Pages.SIGNIN);
+  };
+
+  const validateForm = (values: PasswordValues) => {
+    const errors: Partial<PasswordValues> = {};
+
+    const { newPassword, confirmNewPassword } = values;
+
+    if (!newPassword || !confirmNewPassword) {
+      if (!newPassword) {
+        errors.newPassword = 'A new password is required';
+      }
+      if (!confirmNewPassword) {
+        errors.confirmNewPassword = 'Confirm password is required';
+      }
+    } else if (confirmNewPassword.trim() !== newPassword.trim()) {
+      errors.confirmNewPassword = 'Passwords do not match';
+    }
+
+    return errors;
+  };
+  return (
+    <>
+      <div className="relative flex-col h-full justify-between">
+        <div>
+          <div className="my-4 space-y-4">
+            <p className="text-2xl font-medium text-darkGreen select-none">Reset password</p>
+          </div>
+          <Form
+            validate={validateForm}
+            onSubmit={onSubmit}
+            keepDirtyOnReinitialize
+            render={({ handleSubmit, submitError, submitting, pristine, invalid }) => (
+              <form onSubmit={handleSubmit} autoComplete="false">
+                <div className="space-y-3 w-full">
+                  <Field
+                    name="newPassword"
+                    validate={composeValidators(required, minLength(8), validatePassword)}
```

Comment:
remove validatePassword

Reply:
why?

Reply:
What I meant was, the minLength(8) is good enough for the password validation. This is the validation that the API uses, and the front-end and back-end shouldn't have different validations for password. Either remove minLength or fix the validatePassword function.